DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/mmhost/README.md

```diff
@@ -0,0 +1,363 @@
+    }
+  }
+};
+```
+![](assets/visio-naive.svg)
+
+### Using `annotated_ptr` to configure Avalon memory-mapped host interfaces
+
+You can override the default behaviour of a pointer argument by declaring an `annotated_ptr` kernel argument instead.
+
+To use this feature, you must first make the necessary inclusions
+```c++
+#include <sycl/ext/oneapi/annotated_arg/annotated_ptr.hpp>
+```
+
+#### Example 2: A kernel with a single customized Avalon memory-mapped host interface
+```c++
+struct SingleMMIP {
+  annotated_ptr<int, decltype(properties{
+    register_map,
+    buffer_location<kBL1>,
+    awidth<32>, 
+    dwidth<32>, 
+    latency<0>, 
+    read_write_mode<read>,
+    maxburst<4>
+  })> my_pointer; 
+  void operator()() const {
+    ...
+    // Kernel code
+    my_pointer[x] = y;
+    ...
+  }
+}
+```
+
+The following table describes mutually exclusive properties you can use to customize the interface. Only one may be specified at a time. 
+
+| Parameter                 | Description
+|---                        |---
+| `register_map`            | Pass the pointer for this memory-mapped host interface through the IP component's control/status register
+| `conduit`                 | Pass the pointer for this memory-mapped host interface through a conduit interface 
+
+
+You can use the following parameters to configure your IP component's Avalon memory-mapped host interfaces:
+
+| Parameter                | Default Value | Description
+|---                       |---            |---
+|  `buffer_location<id>`   | N/A           | The address space of the interface that associates with the host. Each unique buffer location will result in a unique Avalon memory-mapped host interface. When `buffer_location` is not specified, then corresponding LSUs will connect to all of the IP's Avalon memory-mapped host interfaces.
+| `awidth<width>`          | 41            | Width of the Avalon memory-mapped host address bus in bits. 
+| `dwidth<width>`          | 64            | Width of the Avalon memory-mapped host data bus in bits. 
+| `latency<value>`         | 0             | Read/Write latency for downstream fixed-latency Avalon memory-mapped agent. For a variable-latency Avalon memory-mapped agent, set `latency<0>`. 
+| `read_write_mode<mode>`  | `read_write`  | Port direction of the interface. (`read_write`, `read` or `write`) 
+| `maxburst<value>`        | 1             | Maximum number of data transfers that can associate with a read or write request. 
+| `alignment<alignment>`   | 1          | Alignment of the Avalon memory-mapped host interface
+| `stable`                 | N/A | User guarantee that the pointer will not change between pipelined invocations of the kernel. The compiler uses this to furthur optimize the kernel.
+
+We can use some of these parameters to improve the performance of `Example 1`. Re-structure the design so that each pointer had exclusive access to a dedicated Avalon memory-mapped agent memory, like this:
```

Comment:
```suggestion
We can use some of these parameters to improve the performance of `Example 1`. Re-structure the design so that each pointer has exclusive access to a dedicated Avalon memory-mapped agent memory, like this:
```